src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/WebRequestSession.cs

```diff
@@ -74,28 +161,95 @@ public class WebRequestSession
         /// </summary>
         public int RetryIntervalInSeconds { get; set; }
 
+        /// <summary>
+        /// Gets or sets the SslProtocol for the session.
+        /// </summary>
+        public WebSslProtocol SslProtocol
+        {
+            get => _sslProtocol; set
+            {
+                if (value != _sslProtocol)
+                {
+                    _changed = true;
+                    _sslProtocol = value;
+                }
+            }
+        }
+
         /// <summary>
         /// Initializes a new instance of the <see cref="WebRequestSession"/> class.
         /// </summary>
         public WebRequestSession()
         {
-            // build the headers collection
+            _changed = true;
+
+            // Build the headers collection
             Headers = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
             ContentHeaders = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
 
-            // build the cookie jar
-            Cookies = new CookieContainer();
+            // Build the cookie jar
+            _cookies = new CookieContainer();
```

Comment:
If I understand the code correctly you can revert _cookies -> Cookies (and the same for all the others in WebRequestSession())

Reply:
You could, but it would go through a longer code path checking for change when we already know it's changed. Is that the preferred option?

Reply:
I think @iSazonov will answer to this question